slack-api-client/src/test/java/test_with_remote_apis/methods/files_Test.java

```diff
@@ -1167,4 +1167,91 @@ public void filesUploadV2_manual_two_files() throws Exception {
+        assertThat(response.getFiles().get(0).getTitle(), is("issue1345 test"));
+    }
+
+    @Test
+    public void issue1345_filesUploadV2_multiple_no_title() throws IOException, SlackApiException {
+        loadRandomChannelId();
+        MethodsClient slackMethods = slack.methods(userToken);
+
+        byte[] fileData = Files.readAllBytes(Paths.get("src/test/resources/sample.txt"));
+
+        // note: this test can be done using only one file as everything tested applies to each file individually - the only important thing is to use "uploadFiles" instead of direct values
+        FilesUploadV2Request.UploadFile uploadFile = FilesUploadV2Request.UploadFile.builder().filename("issue1345_test.txt").fileData(fileData).build();
+        FilesUploadV2Response response = slackMethods.filesUploadV2(r -> r.uploadFiles(List.of(uploadFile)));
+
+        assertThat(response.getError(), is(nullValue()));
+        assertThat(response.getFiles().get(0).getName(), is("issue1345_test.txt"));
+        // title should default to filename
+        assertThat(response.getFiles().get(0).getTitle(), is("issue1345_test.txt"));
+    }
+
+    @Test
+    public void issue1345_filesUploadV2_multiple_no_filename() throws IOException, SlackApiException {
+        loadRandomChannelId();
+        MethodsClient slackMethods = slack.methods(userToken);
+
+        byte[] fileData = Files.readAllBytes(Paths.get("src/test/resources/sample.txt"));
+
+        // note: this test can be done using only one file as everything tested applies to each file individually - the only important thing is to use "uploadFiles" instead of direct values
+        FilesUploadV2Request.UploadFile uploadFile = FilesUploadV2Request.UploadFile.builder().title("issue1345 test").fileData(fileData).build();
+        FilesUploadV2Response response = slackMethods.filesUploadV2(r -> r.uploadFiles(List.of(uploadFile)));
+
+        assertThat(response.getError(), is(nullValue()));
+        // filename defaults to "Uploaded file", which will be converted by slack servers to be lowercase only, contain no special characters but underscores, dots and dashes, etc.
+        assertThat(response.getFiles().get(0).getName(), is("uploaded_file"));
+        // title should be unaffected if it has a value set
+        assertThat(response.getFiles().get(0).getTitle(), is("issue1345 test"));
+    }
+
+    @Test
+    public void issue1345_filesUploadV2_multiple_no_filename_no_title_file() throws IOException, SlackApiException {
+        loadRandomChannelId();
+        MethodsClient slackMethods = slack.methods(userToken);
+
+        File file = new File("src/test/resources/sample.txt");
+
+        // note: this test can be done using only one file as everything tested applies to each file individually - the only important thing is to use "uploadFiles" instead of direct values
+        FilesUploadV2Request.UploadFile uploadFile = FilesUploadV2Request.UploadFile.builder().file(fileData).build();
```

Comment:
```suggestion
        FilesUploadV2Request.UploadFile uploadFile = FilesUploadV2Request.UploadFile.builder().file(file).build();
```

One other change to test! 🙏